cdap-app-fabric/src/main/java/co/cask/cdap/security/DefaultUGIProvider.java

```diff
@@ -87,6 +88,8 @@ public UserGroupInformation getConfiguredUGI(ImpersonationInfo impersonationInfo
       String expandedPrincipal = SecurityUtil.expandPrincipal(impersonationInfo.getPrincipal());
       LOG.debug("Logging in as: principal={}, keytab={}", expandedPrincipal, localKeytabFile);
 
+      FileUtils.assertReadableFile(localKeytabFile);
```

Comment:
From javadoc of `Files.isReadable`:

`true if the file exists and is readable; false if the file does not exist, read access would be denied because the Java virtual machine has insufficient privileges, or access cannot be determined`